ocaml/xapi/gpg.ml

```diff
@@ -152,3 +152,91 @@ let with_detached_signature filename signature size f =
 
 let with_verified_signature filename signature f =
   common `verified_signature filename signature Int64.zero f
+
+let assert_name_is_valid ~name =
+  let safe_char = function
+    | 'A' .. 'Z' | 'a' .. 'z' | '0' .. '9' | '.' | '_' | '-' ->
+        true
+    | _ ->
+        false
+  in
+  let len = String.length name in
+  let starts_with_dot = Astring.String.is_prefix ~affix:"." name in
+  let is_valid = String.for_all safe_char name && not starts_with_dot in
+  match (is_valid, len <= 0, len > 64) with
+  | false, _, _ ->
+      raise Api_errors.(Server_error (gpg_key_name_is_invalid, [name]))
+  | true, true, _ ->
+      raise Api_errors.(Server_error (gpg_key_name_is_empty, []))
+  | true, false, true ->
+      raise Api_errors.(Server_error (gpg_key_name_is_too_long, [name]))
+  | true, false, false ->
+      ()
+
+module PubKeyMetaData = struct
+  type t = {created: float; fingerprint: string}
```

Comment:
Could you change the created field to be Ptime.t? This makes it obvious it's a unit timestamp. This needs an awkward conversion from ptime to float to date to store it in the database, I'm fixing this is https://github.com/xapi-project/stdext/pull/68/

Reply:
Sure. I will change it once the PR got merged.